### ISSUE TITLE
Question about the require syntax

### ISSUE THREAD

**Issue by brynne8** [Stepets/utf8.lua#8]
I want to ask what's the difference between
```lua
local utf8 = require('.utf8'):init()
```
and
```lua
local utf8 = require('utf8'):init()
```

I placed this library in my package path in the folder `utf8` (using LuaJIT), when I run my test, `utf8.codepoint` is undefined. Where should I place this library to get `.utf8` work.

**Comment by Stepets** [comment]
For LuaJIT Lua5.1 Lua5.2 specifying leading dot is not necessary.
As mentioned in the readme Lua5.3 has default builtin utf8 module (package.loaded.utf8). So for Lua5.3 require(".utf8") is a workaround to allow search in file-system. Dot is a placeholder for system-specific path separator ("/" on posix, "\" on windows).
Detailed module loading procedure is described here https://www.lua.org/manual/5.3/manual.html#6.3

Although my Lua5.3 detection was wrong and some functions can be not initialized.
Fixed in 87acef6ffe8ee75f1e3c08cac4932132611b3ebc

**Comment by brynne8** [comment]
Thanks.
